src/osd/PrimaryLogPG.cc

```diff
@@ -5798,11 +5798,20 @@ int PrimaryLogPG::do_extent_cmp(OpContext *ctx, OSDOp& osd_op)
 
 int PrimaryLogPG::finish_extent_cmp(OSDOp& osd_op, const bufferlist &read_bl)
 {
-  for (uint64_t idx = 0; idx < osd_op.indata.length(); ++idx) {
-    char read_byte = (idx < read_bl.length() ? read_bl[idx] : 0);
-    if (osd_op.indata[idx] != read_byte) {
-        return (-MAX_ERRNO - idx);
-    }
+  auto input_iter = osd_op.indata.begin();
+  auto read_iter = read_bl.begin();
+  uint64_t idx = 0;
+
+  while (input_iter != osd_op.indata.end()) {
+      char read_byte = (read_iter != read_bl.end() ? *read_iter : 0);
```

Comment:
Nit: Indentation is off here and in the if statements below.  The indent should be 2 spaces.

Reply:
updated

Reply:
It's still off in the if statements.  It should be
```
    if (*input_iter != read_byte) {
      return (-MAX_ERRNO - idx);
    }
```
```
    if (read_iter != read_bl.end()) {
      ++read_iter;
    }
```

Reply:
thanks, updated